### ISSUE TITLE
ExpandoObject with collections

### ISSUE THREAD

**Issue by Thowk** [ValeraT1982/ObjectsComparer#2]
It seems to be a problem when comparing ExpandoObject containing collections. 

I try to compare such jsons:

`
{
  "Transaction": [
    {
      "Name": "abc",
      "No": 101
    }
  ]
}
`
`
{
  "Transaction": [
    {
      "Name": "abc",
      "No": 102
    }
  ]
}
`

This is how my configuration looks like:
``` cs
var comparer = new Comparer(new ComparisonSettings { RecursiveComparison = true });

dynamic responseJson = JsonConvert.DeserializeObject<ExpandoObject>(
    File.ReadAllText("result1.json"));

dynamic expectedJson = JsonConvert.DeserializeObject<ExpandoObject>(
    File.ReadAllText("result2.json"));

var isEqual = comparer.Compare(responseJson, expectedJson, out IEnumerable<Difference> diff);
```
Compare returns 'true'. (JsonConvert.DeserializeObject comes from Newtonsoft.Json)

Do you know how to solve that problem (is it a bug, or not yet implemented functionality)?

**Comment by GregFinzer** [comment]
Is this for .net core?

**Comment by Thowk** [comment]
Yes, but I guess the same is on i.e net461
________________________________

**Comment by ValeraT1982** [comment]
Sorry for later answer. Every time I go to vacation somebody has an issue :)
It doesn't work because the type of "Transaction" property is "System.Collections.Generic.List<object>", so Comparer compares every element as an object (object doesn't have properties No and Name).
Fix for this issue is committed now and probably will be in the next version (I need to think if it affects any other cases) in September.

@Thowk, you can download solution and build it manually to have this issue fixed or wait for the next version to be released.

**Comment by ValeraT1982** [comment]
Resolved in 1.2
